src/AudioAnalysisTools/LongDurationSpectrograms/LDSpectrogramRGB.cs

```diff
@@ -540,10 +545,38 @@ public void DrawGreyScaleSpectrograms(DirectoryInfo opdir, string opFileName, st
                     continue;
                 }
 
-                // the directory for the following path must exist
-                var path = FilenameHelpers.AnalysisResultPath(opdir, opFileName, key, "png");
                 var bmp = this.DrawGreyscaleSpectrogramOfIndex(key);
-                bmp?.Save(path);
+
+                var header = new Bitmap(bmp.Width, 20);
+                Graphics g = Graphics.FromImage(header);
+                g.Clear(Color.LightGray);
+                g.SmoothingMode = SmoothingMode.AntiAlias;
+                //g.InterpolationMode = InterpolationMode.HighQualityBicubic;
+                //g.PixelOffsetMode = PixelOffsetMode.HighQuality;
+                g.DrawString(key, new Font("Tahoma", 9), Brushes.Black, 4, 4);
+                var indexImage = ImageTools.CombineImagesVertically(new List<Image>(new Image[] { header, bmp }));
+
+                // save the image - the directory for the path must exist
+                //var path = FilenameHelpers.AnalysisResultPath(opdir, opFileName, key, "png");
+                //indexImage?.Save(path);
+
+                listOfImages.Add(indexImage);
+            }
+
+            // save the combined track image - the directory for the path must exist
+            var path = FilenameHelpers.AnalysisResultPath(opdir, opFileName, "KEYS", "png");
+            int imageCount = listOfImages.Count;
+
+            // check how wide combined image will be. If tracks are wider than 180 = 3 hours, then go vertical
+            if (listOfImages[0].Width * imageCount > 180 * imageCount)
+            {
+                var combinedImage = ImageTools.CombineImagesVertically(listOfImages);
+                combinedImage?.Save(path);
+            }
+            else
+            {
+                var combinedImage = ImageTools.CombineImagesInLine(listOfImages);
+                combinedImage?.Save(path);
```

Comment:
This is not OK. We never want to generate inconsistent output.

Our images always should have the same format and structure, otherwise, it is hard to script and use the results.